File: javascript/problems/twoSum2.js
function twoSum(numbers, target) {
  let lo = 0;
  let hi = numbers.length - 1;
  let sum;
  while (lo < hi) {
    sum = numbers[lo] + numbers[hi];
    if (sum === target) {
      return [lo+1, hi+1]
    } else if (sum < target) {
      lo++;
    } else {
      hi--;
    }
  }
  return [];
}

console.log(twoSum([2,7,11,15], 9));
console.log(twoSum([2,3,4], 6));
console.log(twoSum([2,7,11,15], 2));
console.log(twoSum([-1,0], -1));

File: golang/exercise/two_sum_2.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	lo, hi := 0, len(nums)-1
	for lo < hi {
		sum := nums[lo] + nums[hi]
		if sum == target {
			return []int{lo, hi}
		}
		if sum < target {
			lo++
		} else {
			hi--
		}
	}
	return nil
}

func main() {
	fmt.Println(twoSum([]int{2, 7, 11, 15}, 9))
	fmt.Println(twoSum([]int{2, 3, 4}, 6))
	fmt.Println(twoSum([]int{2, 7, 11, 15}, 2))
	fmt.Println(twoSum([]int{-1, 0}, -1))
}
